### PR TITLE
Add SAI_INGRESS_PRIORITY_GROUP_STAT_DROPPED_PACKETS counter, create new FlexCounter group

### PR DESCRIPTION
**Pull request by ayurkiv-nvda**
Signed-off-by: Andriy Yurkiv <ayurkiv@nvidia.com>

**depends on :**
https://github.com/Azure/sonic-utilities/pull/1355
https://github.com/Azure/sonic-buildimage/pull/6444

**merge priority:**
https://github.com/Azure/sonic-swss/pull/1600
https://github.com/Azure/sonic-utilities/pull/1355
https://github.com/Azure/sonic-buildimage/pull/6444

**What I did**
Added new option for "counterpoll" util

**Why I did it**
Add appropriate code to counterpoll/main.py

**How I verified it**
```
admin@arc-switch1041:~$ counterpoll pg-drop enable  --> enable new counter
admin@arc-switch1041:~$ counterpoll show  --> check new PG_DROP_STAT  counter status
```
check counters
```
admin@arc-switch1041:~$ redis-cli -n 2
127.0.0.1:6379[2]> HGETALL COUNTERS:oid:0x1a000000000062
1) "SAI_INGRESS_PRIORITY_GROUP_STAT_XOFF_ROOM_WATERMARK_BYTES"
2) "0"
3) "SAI_INGRESS_PRIORITY_GROUP_STAT_SHARED_WATERMARK_BYTES"
4) "0"
5) "SAI_INGRESS_PRIORITY_GROUP_STAT_DROPPED_PACKETS"
6) "0"
```
**Details if related**

### CODE REVIEW

**Comment by volodymyrsamotiy** [comment]
@neethajohn, could you you please review?

**Comment by liat-grozovik** [comment]
@neethajohn if this is needed for 201911 please add the proper label

**Review by neethajohn — COMMENTED** [review]
Please add tests for the flex counter group/table creation/deletion. You can take a look at tests/test_drop_counters.py

File: orchagent/flexcounterorch.cpp [hunk 1]
```diff
@@ -26,6 +26,7 @@ unordered_map<string, string> flexCounterGroupMap =
     {"PFCWD", PFC_WD_FLEX_COUNTER_GROUP},
     {"QUEUE_WATERMARK", QUEUE_WATERMARK_STAT_COUNTER_FLEX_COUNTER_GROUP},
     {"PG_WATERMARK", PG_WATERMARK_STAT_COUNTER_FLEX_COUNTER_GROUP},
+    {"PG_DROP", PG_DROPPED_PACKETS_STAT_COUNTER_FLEX_COUNTER_GROUP},
```
[the remark this answers is not in the document]
Comment (ayurkiv-nvda) [reply orchagent/flexcounterorch.cpp]: Done

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -198,6 +199,11 @@ static const vector<sai_ingress_priority_group_stat_t> ingressPriorityGroupWater
     SAI_INGRESS_PRIORITY_GROUP_STAT_SHARED_WATERMARK_BYTES,
 };
 
+static const vector<sai_ingress_priority_group_stat_t> ingressPriorityGroupDroppedPktsStatIds =
```
[the remark this answers is not in the document]
Comment (ayurkiv-nvda) [reply orchagent/portsorch.cpp]: Done

File: orchagent/portsorch.h [hunk 3]
```diff
@@ -22,7 +22,7 @@
 #define QUEUE_STAT_COUNTER_FLEX_COUNTER_GROUP "QUEUE_STAT_COUNTER"
 #define QUEUE_WATERMARK_STAT_COUNTER_FLEX_COUNTER_GROUP "QUEUE_WATERMARK_STAT_COUNTER"
 #define PG_WATERMARK_STAT_COUNTER_FLEX_COUNTER_GROUP "PG_WATERMARK_STAT_COUNTER"
-
+#define PG_DROPPED_PACKETS_STAT_COUNTER_FLEX_COUNTER_GROUP "PG_DROPPED_PACKETS_STAT_COUNTER"
```
[the remark this answers is not in the document]
Comment (ayurkiv-nvda) [reply orchagent/portsorch.h]: Done

File: orchagent/portsorch.cpp [hunk 4]
```diff
@@ -55,6 +55,7 @@ extern Directory<Orch*> gDirectory;
 #define QUEUE_STAT_FLEX_COUNTER_POLLING_INTERVAL_MS   10000
 #define QUEUE_WATERMARK_FLEX_STAT_COUNTER_POLL_MSECS "10000"
 #define PG_WATERMARK_FLEX_STAT_COUNTER_POLL_MSECS    "10000"
+#define PG_DROPPED_PACKETS_FLEX_STAT_COUNTER_POLL_MSECS "10000"
```
[the remark this answers is not in the document]
Comment (ayurkiv-nvda) [reply orchagent/portsorch.cpp]: Done

**Comment by liat-grozovik** [comment]
@neethajohn  could you please review recent changes?

**Comment by neethajohn** [comment]
> Please add tests for the flex counter group/table creation/deletion. You can take a look at tests/test_drop_counters.py

No unit tests have been added. Please add

**Comment by ayurkiv-nvda** [comment]
> Please add tests for the flex counter group/table creation/deletion. You can take a look at tests/test_drop_counters.py

tests in "tests/test_drop_counters.py" are different.  We do not expect dynamic creation and deletion tables in this functionality.
I think tests in tests/test_watermark.py is more suitable for example

**Comment by ayurkiv-nvda** [comment]
> Please add tests for the flex counter group/table creation/deletion. You can take a look at tests/test_drop_counters.py

test added

**Review by neethajohn — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@ayurkiv-nvda can you please update the swss submodule pointer so we can move forward with the rest?

**Comment by liat-grozovik** [comment]
@neethajohn  please add label 'request for 202012' i am not able to add and this is needed.

**Comment by abdosi** [comment]
@ayurkiv-nvda can ypu please create PR for 201911 ? Cherry-pick has conflict

**Pull request merged**